Index: src/md2/md2_one.c
===================================================================
--- src/md2/md2_one.c
+++ src/md2/md2_one.c
@@ -35,3 +35,8 @@ unsigned char *xdig_md2(const unsigned char *d, si

     return digest;
 }
+
+size_t xdig_md2_ctx_size(void)
+{
+    return sizeof(xdig_md2_ctx_t);
+}
Index: src/md4/md4_one.c
===================================================================
--- src/md4/md4_one.c
+++ src/md4/md4_one.c
@@ -35,3 +35,8 @@ unsigned char *xdig_md4(const unsigned char *d, si

     return digest;
 }
+
+size_t xdig_md4_ctx_size(void)
+{
+    return sizeof(xdig_md4_ctx_t);
+}
Index: src/md5/md5_one.c
===================================================================
--- src/md5/md5_one.c
+++ src/md5/md5_one.c
@@ -35,3 +35,8 @@ unsigned char *xdig_md5(const unsigned char *d, si

     return digest;
 }
+
+size_t xdig_md5_ctx_size(void)
+{
+    return sizeof(xdig_md5_ctx_t);
+}
Index: src/sha/sha256.c
===================================================================
--- src/sha/sha256.c
+++ src/sha/sha256.c
@@ -101,6 +101,16 @@ unsigned char *xdig_sha224(const unsigned char *d,
     return md;
 }
 
+size_t xdig_sha256_ctx_size(void)
+{
+    return sizeof(xdig_sha256_ctx_t);
+}
+
+size_t xdig_sha224_ctx_size(void)
+{
+    return xdig_sha256_ctx_size();
+}
+
 #define DATA_ORDER_IS_BIG_ENDIAN
 
 #define HASH_LONG               XDIG_SHA_LONG
Index: src/sha/sha512.c
===================================================================
--- src/sha/sha512.c
+++ src/sha/sha512.c
@@ -177,6 +177,16 @@ unsigned char *xdig_sha512(const unsigned char *d,
     return md;
 }
 
+size_t xdig_sha512_ctx_size(void)
+{
+    return sizeof(xdig_sha512_ctx_t);
+}
+
+size_t xdig_sha384_ctx_size(void)
+{
+    return xdig_sha512_ctx_size();
+}
+
 #ifndef SHA512_ASM
 static
 #else
