Index: xdigest/core/cpuid.c
===================================================================
--- xdigest/core/cpuid.c	(revision 6114)
+++ xdigest/core/cpuid.c	(working copy)
@@ -8,6 +8,7 @@
  */
 
 #include "internal/e_os.h"
 /* ignored include 'crypto/cryptlib.h' */
+#include "xdigest/xdigest.h"
 
 #if     defined(__i386)   || defined(__i386__)   || defined(_M_IX86) || \
         defined(__x86_64) || defined(__x86_64__) || \
@@ -171,3 +172,9 @@ size_t xdig_instrument_bus2(unsigned int *out, si
     return 0;
 }
+
+/* Public interface. */
+void xdig_init()
+{
+    xdig_cpuid_setup();
+}
 #endif
Index: xdigest/core/armcap.c
===================================================================
--- xdigest/core/armcap.c
+++ xdigest/core/armcap.c
@@ -448,3 +448,9 @@ void xdig_cpuid_setup(void)
 # endif
 }
 #endif /* _WIN32, __ARM_MAX_ARCH__ >= 7 */
+
+/* Public interface. */
+void xdig_init()
+{
+    xdig_cpuid_setup();
+}
