Index: src/md2/md2_one.c
===================================================================
--- src/md2/md2_one.c
+++ src/md2/md2_one.c
@@ -23,32 +23,17 @@
  * functions to different names
  */
 
-unsigned char *xdig_md2(const unsigned char *d, size_t n, unsigned char *md)
+unsigned char *
+xdig_md2(const unsigned char *data, size_t len, unsigned char *digest)
 {
-    xdig_md2_ctx_t c;
-    static unsigned char m[XDIG_MD2_DIGEST_LENGTH];
+    xdig_md2_ctx_t ctx;
 
-    if (md == NULL)
-        md = m;
-    if (!xdig_md2_init(&c))
-        return NULL;
-#ifndef CHARSET_EBCDIC
-    xdig_md2_update(&c, d, n);
-#else
-    {
-        char temp[1024];
-        unsigned long chunk;
+    xdig_md2_init(&ctx);
+    xdig_md2_update(&ctx, data, len);
+    xdig_md2_final(digest, &ctx);
+    xdig_cleanse(&ctx, sizeof(ctx)); /* Security consideration */
 
-        while (n > 0) {
-            chunk = (n > sizeof(temp)) ? sizeof(temp) : n;
-            ebcdic2ascii(temp, d, chunk);
-            xdig_md2_update(&c, temp, chunk);
-            n -= chunk;
-            d += chunk;
-        }
-    }
-#endif
-    xdig_md2_final(md, &c);
-    xdig_cleanse(&c, sizeof(c)); /* Security consideration */
-    return md;
+    return digest;
 }
