Index: xdigest/md4/md4_one.c
===================================================================
--- xdigest/md4/md4_one.c
+++ xdigest/md4/md4_one.c
@@ -23,32 +23,17 @@
 /* ignored include 'openssl/ebcdic.h' */
 #endif
 
-unsigned char *xdig_md4(const unsigned char *d, size_t n, unsigned char *md)
+unsigned char *
+xdig_md4(const unsigned char *data, size_t len, unsigned char *digest)
 {
-    xdig_md4_ctx_t c;
-    static unsigned char m[XDIG_MD4_DIGEST_LENGTH];
+    xdig_md4_ctx_t ctx;
 
-    if (md == NULL)
-        md = m;
-    if (!xdig_md4_ctx_init(&c))
-        return NULL;
-#ifndef CHARSET_EBCDIC
-    xdig_md4_ctx_update(&c, d, n);
-#else
-    {
-        char temp[1024];
-        unsigned long chunk;
+    xdig_md4_ctx_init(&ctx);
+    xdig_md4_ctx_update(&ctx, data, len);
+    xdig_md4_ctx_final(digest, &ctx);
+    xdig_cleanse(&ctx, sizeof(ctx)); /* security consideration */
 
-        while (n > 0) {
-            chunk = (n > sizeof(temp)) ? sizeof(temp) : n;
-            ebcdic2ascii(temp, d, chunk);
-            xdig_md4_ctx_update(&c, temp, chunk);
-            n -= chunk;
-            d += chunk;
-        }
-    }
-#endif
-    xdig_md4_ctx_final(md, &c);
-    xdig_cleanse(&c, sizeof(c)); /* security consideration */
-    return md;
+    return digest;
 }
