Index: src/core/mem_clr.c
===================================================================
--- src/core/mem_clr.c	(revision 6199)
+++ src/core/mem_clr.c	(working copy)
@@ -10,16 +10,7 @@
 #include <string.h>
 /* ignored include 'openssl/crypto.h' */
 
-/*
- * Pointer to memset is volatile so that compiler must de-reference
- * the pointer and can't assume that it points to any function in
- * particular (such as memset, which it then might further "optimize")
- */
-typedef void *(*memset_t)(void *, int, size_t);
-
-static volatile memset_t memset_func = memset;
-
 void xhash_cleanse(void *ptr, size_t len)
 {
-    memset_func(ptr, 0, len);
+    memset(ptr, 0, len);
 }
