Index: include/crypto/md32_common.h
===================================================================
--- include/crypto/md32_common.h        (revision 6040)
+++ include/crypto/md32_common.h        (working copy)
@@ -67,7 +67,6 @@
 # define OSSL_CRYPTO_MD32_COMMON_H
 # pragma once
 
-# include <openssl/crypto.h>
 /*
  * For ossl_(un)likely
  */
Index: include/internal/common.h
===================================================================
--- include/internal/common.h   (revision 6040)
+++ include/internal/common.h   (working copy)
@@ -16,7 +16,6 @@
 # include "openssl/configuration.h"
 
 # include "internal/e_os.h" /* ossl_inline in many files */
-# include "internal/nelem.h"
 
 # if defined(__GNUC__) || defined(__clang__)
 #  define ossl_likely(x)     __builtin_expect(!!(x), 1)
Index: include/internal/e_os.h
===================================================================
--- include/internal/e_os.h     (revision 6040)
+++ include/internal/e_os.h     (working copy)
@@ -11,10 +11,8 @@
 # define OSSL_E_OS_H
 
 # include <limits.h>
-# include <openssl/opensslconf.h>
 
 # include <openssl/e_os2.h>
-# include <openssl/crypto.h>
 # include "internal/numbers.h"   /* Ensure the definition of SIZE_MAX */
