Index: src/sha/sha256.c
===================================================================
--- src/sha/sha256.c
+++ src/sha/sha256.c
@@ -7,20 +7,10 @@
  * https://www.openssl.org/source/license.html
  */
 
-/*
- * SHA256 low level APIs are deprecated for public use, but still ok for
- * internal use.
- */
-#include "internal/deprecated.h"
-
-#include <openssl/opensslconf.h>
-
 #include <stdlib.h>
 #include <string.h>
 
-#include <openssl/crypto.h>
 #include <openssl/sha.h>
-#include <openssl/opensslv.h>
 #include "internal/endian.h"
 #include "crypto/sha.h"
 
Index: src/sha/sha512.c
===================================================================
--- src/sha/sha512.c
+++ src/sha/sha512.c
@@ -11,10 +11,7 @@
  * SHA512 low level APIs are deprecated for public use, but still ok for
  * internal use.
  */
-#include "internal/deprecated.h"
-
 #include <stdio.h>
-#include <openssl/opensslconf.h>
 /*-
  * IMPLEMENTATION NOTES.
  *
@@ -52,11 +49,8 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <openssl/crypto.h>
 #include <openssl/sha.h>
-#include <openssl/opensslv.h>
 
-#include "internal/cryptlib.h"
 #include "crypto/sha.h"
 
 #if defined(__i386) || defined(__i386__) || defined(_M_IX86) || \
Index: src/sha/sha_local.h
===================================================================
--- src/sha/sha_local.h
+++ src/sha/sha_local.h
@@ -10,7 +10,6 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <openssl/opensslconf.h>
 #include <openssl/sha.h>
 #include "internal/endian.h"
Index: src/cpuid.c
===================================================================
--- src/cpuid.c
+++ src/cpuid.c
@@ -8,7 +8,6 @@
  */
 
 #include "internal/e_os.h"
-#include "crypto/cryptlib.h"
 
 #if     defined(__i386)   || defined(__i386__)   || defined(_M_IX86) || \
         defined(__x86_64) || defined(__x86_64__) || \
