Index: src/sha/sha256.c
===================================================================
--- src/sha/sha256.c
+++ src/sha/sha256.c
@@ -104,9 +104,9 @@ int xhash_sha224_final(unsigned char *md, xhash_sh
         }                               \
     } while (0)
 
-#define HASH_UPDATE             SHA256_Update
-#define HASH_TRANSFORM          SHA256_Transform
-#define HASH_FINAL              SHA256_Final
+#define HASH_UPDATE             xhash_sha256_update
+#define HASH_TRANSFORM          xhash_sha256_transform
+#define HASH_FINAL              xhash_sha256_final
 #define HASH_BLOCK_DATA_ORDER   sha256_block_data_order
 #ifndef SHA256_ASM
 static
