Index: include/xhash/sha.h
===================================================================
--- include/xhash/sha.h
+++ include/xhash/sha.h
@@ -45,10 +45,10 @@ typedef struct SHAstate_st {
     unsigned int num;
 } SHA_CTX;
 
-OSSL_DEPRECATEDIN_3_0 int xhash_sha1_init(SHA_CTX *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha1_update(SHA_CTX *c, const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha1_final(unsigned char *md, SHA_CTX *c);
-OSSL_DEPRECATEDIN_3_0 void xhash_sha1_transform(SHA_CTX *c, const unsigned char *data);
+int xhash_sha1_init(SHA_CTX *c);
+int xhash_sha1_update(SHA_CTX *c, const void *data, size_t len);
+int xhash_sha1_final(unsigned char *md, SHA_CTX *c);
+void xhash_sha1_transform(SHA_CTX *c, const unsigned char *data);
 # endif
 
 unsigned char *SHA1(const unsigned char *d, size_t n, unsigned char *md);
@@ -65,16 +65,13 @@ typedef struct SHA256state_st {
     unsigned int num, md_len;
 } SHA256_CTX;
 
-OSSL_DEPRECATEDIN_3_0 int xhash_sha224_init(SHA256_CTX *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha224_update(SHA256_CTX *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha224_final(unsigned char *md, SHA256_CTX *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha256_init(SHA256_CTX *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha256_update(SHA256_CTX *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha256_final(unsigned char *md, SHA256_CTX *c);
-OSSL_DEPRECATEDIN_3_0 void xhash_sha256_transform(SHA256_CTX *c,
-                                            const unsigned char *data);
+int xhash_sha224_init(SHA256_CTX *c);
+int xhash_sha224_update(SHA256_CTX *c, const void *data, size_t len);
+int xhash_sha224_final(unsigned char *md, SHA256_CTX *c);
+int xhash_sha256_init(SHA256_CTX *c);
+int xhash_sha256_update(SHA256_CTX *c, const void *data, size_t len);
+int xhash_sha256_final(unsigned char *md, SHA256_CTX *c);
+void xhash_sha256_transform(SHA256_CTX *c, const unsigned char *data);
 # endif
 
 unsigned char *SHA224(const unsigned char *d, size_t n, unsigned char *md);
@@ -116,16 +113,13 @@ typedef struct SHA512state_st {
     unsigned int num, md_len;
 } SHA512_CTX;
 
-OSSL_DEPRECATEDIN_3_0 int xhash_sha384_init(SHA512_CTX *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha384_update(SHA512_CTX *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha384_final(unsigned char *md, SHA512_CTX *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha512_init(SHA512_CTX *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha512_update(SHA512_CTX *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha512_final(unsigned char *md, SHA512_CTX *c);
-OSSL_DEPRECATEDIN_3_0 void xhash_sha512_transform(SHA512_CTX *c,
-                                            const unsigned char *data);
+int xhash_sha384_init(SHA512_CTX *c);
+int xhash_sha384_update(SHA512_CTX *c, const void *data, size_t len);
+int xhash_sha384_final(unsigned char *md, SHA512_CTX *c);
+int xhash_sha512_init(SHA512_CTX *c);
+int xhash_sha512_update(SHA512_CTX *c, const void *data, size_t len);
+int xhash_sha512_final(unsigned char *md, SHA512_CTX *c);
+void xhash_sha512_transform(SHA512_CTX *c, const unsigned char *data);
 # endif
 
 unsigned char *SHA384(const unsigned char *d, size_t n, unsigned char *md);
