Index: include/xhash/sha.h
===================================================================
--- include/xhash/sha.h
+++ include/xhash/sha.h
@@ -11,10 +11,6 @@
 # define xhash_SHA_H
 # pragma once
 
-# ifndef xhash_NO_DEPRECATED_3_0
-#  define HEADER_SHA_H
-# endif
-
 # include <xhash/e_os2.h>
 # include <stddef.h>
 
@@ -24,7 +20,6 @@ extern "C" {
 
 # define SHA_DIGEST_LENGTH 20
 
-# ifndef xhash_NO_DEPRECATED_3_0
 /*-
  * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
  * ! SHA_LONG has to be at least 32 bits wide.                    !
@@ -49,11 +44,9 @@ int xhash_sha1_init(xhash_sha1_ctx_t *c);
 int xhash_sha1_update(xhash_sha1_ctx_t *c, const void *data, size_t len);
 int xhash_sha1_final(unsigned char *md, xhash_sha1_ctx_t *c);
 void xhash_sha1_transform(xhash_sha1_ctx_t *c, const unsigned char *data);
-# endif
 
 unsigned char *xhash_sha1(const unsigned char *d, size_t n, unsigned char *md);
 
-# ifndef xhash_NO_DEPRECATED_3_0
 #  define SHA256_CBLOCK   (SHA_LBLOCK*4)/* SHA-256 treats input data as a
                                         * contiguous array of 32 bit wide
                                         * big-endian values. */
@@ -72,7 +65,6 @@ int xhash_sha256_init(xhash_sha256_ctx_t *c);
 int xhash_sha256_update(xhash_sha256_ctx_t *c, const void *data, size_t len);
 int xhash_sha256_final(unsigned char *md, xhash_sha256_ctx_t *c);
 void xhash_sha256_transform(xhash_sha256_ctx_t *c, const unsigned char *data);
-# endif
 
 unsigned char *xhash_sha224(const unsigned char *d, size_t n, unsigned char *md);
 unsigned char *xhash_sha256(const unsigned char *d, size_t n, unsigned char *md);
@@ -83,7 +75,6 @@ unsigned char *xhash_sha256(const unsigned char *d
 # define SHA384_DIGEST_LENGTH    48
 # define SHA512_DIGEST_LENGTH    64
 
-# ifndef xhash_NO_DEPRECATED_3_0
 /*
  * Unlike 32-bit digest algorithms, SHA-512 *relies* on SHA_LONG64
  * being exactly 64-bit wide. See Implementation Notes in sha512.c
@@ -128,5 +119,3 @@ unsigned char *xhash_sha512(const unsigned char *d
 # ifdef  __cplusplus
 }
 # endif
-
-#endif
