Index: include/xhash/sha.h
===================================================================
--- include/xhash/sha.h
+++ include/xhash/sha.h
@@ -45,10 +45,10 @@ typedef struct SHAstate_st {
     unsigned int num;
 } xhash_sha1_ctx_t;
 
-OSSL_DEPRECATEDIN_3_0 int xhash_sha1_init(xhash_sha1_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha1_update(xhash_sha1_ctx_t *c, const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha1_final(unsigned char *md, xhash_sha1_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 void xhash_sha1_transform(xhash_sha1_ctx_t *c, const unsigned char *data);
+int xhash_sha1_init(xhash_sha1_ctx_t *c);
+int xhash_sha1_update(xhash_sha1_ctx_t *c, const void *data, size_t len);
+int xhash_sha1_final(unsigned char *md, xhash_sha1_ctx_t *c);
+void xhash_sha1_transform(xhash_sha1_ctx_t *c, const unsigned char *data);
 # endif
 
 unsigned char *SHA1(const unsigned char *d, size_t n, unsigned char *md);
@@ -65,16 +65,13 @@ typedef struct SHA256state_st {
     unsigned int num, md_len;
 } xhash_sha256_ctx_t;
 
-OSSL_DEPRECATEDIN_3_0 int xhash_sha224_init(xhash_sha256_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha224_update(xhash_sha256_ctx_t *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha224_final(unsigned char *md, xhash_sha256_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha256_init(xhash_sha256_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha256_update(xhash_sha256_ctx_t *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha256_final(unsigned char *md, xhash_sha256_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 void xhash_sha256_transform(xhash_sha256_ctx_t *c,
-                                            const unsigned char *data);
+int xhash_sha224_init(xhash_sha256_ctx_t *c);
+int xhash_sha224_update(xhash_sha256_ctx_t *c, const void *data, size_t len);
+int xhash_sha224_final(unsigned char *md, xhash_sha256_ctx_t *c);
+int xhash_sha256_init(xhash_sha256_ctx_t *c);
+int xhash_sha256_update(xhash_sha256_ctx_t *c, const void *data, size_t len);
+int xhash_sha256_final(unsigned char *md, xhash_sha256_ctx_t *c);
+void xhash_sha256_transform(xhash_sha256_ctx_t *c, const unsigned char *data);
 # endif
 
 unsigned char *SHA224(const unsigned char *d, size_t n, unsigned char *md);
@@ -116,16 +113,13 @@ typedef struct SHA512state_st {
     unsigned int num, md_len;
 } xhash_sha512_ctx_t;
 
-OSSL_DEPRECATEDIN_3_0 int xhash_sha384_init(xhash_sha512_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha384_update(xhash_sha512_ctx_t *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha384_final(unsigned char *md, xhash_sha512_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha512_init(xhash_sha512_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha512_update(xhash_sha512_ctx_t *c,
-                                        const void *data, size_t len);
-OSSL_DEPRECATEDIN_3_0 int xhash_sha512_final(unsigned char *md, xhash_sha512_ctx_t *c);
-OSSL_DEPRECATEDIN_3_0 void xhash_sha512_transform(xhash_sha512_ctx_t *c,
-                                            const unsigned char *data);
+int xhash_sha384_init(xhash_sha512_ctx_t *c);
+int xhash_sha384_update(xhash_sha512_ctx_t *c, const void *data, size_t len);
+int xhash_sha384_final(unsigned char *md, xhash_sha512_ctx_t *c);
+int xhash_sha512_init(xhash_sha512_ctx_t *c);
+int xhash_sha512_update(xhash_sha512_ctx_t *c, const void *data, size_t len);
+int xhash_sha512_final(unsigned char *md, xhash_sha512_ctx_t *c);
+void xhash_sha512_transform(xhash_sha512_ctx_t *c, const unsigned char *data);
 # endif
 
 unsigned char *SHA384(const unsigned char *d, size_t n, unsigned char *md);
