Index: include/xhash/sha.h
===================================================================
--- include/xhash/sha.h	(revision 6136)
+++ include/xhash/sha.h	(working copy)
@@ -11,11 +11,6 @@
 # define xhash_SHA_H
 # pragma once
 
-/* ignored include 'openssl/macros.h' */
-# ifndef xhash_NO_DEPRECATED_3_0
-#  define HEADER_SHA_H
-# endif
-
 #include "internal/e_os2.h"
 # include <stddef.h>
 
@@ -25,7 +20,6 @@ extern "C" {
 
 # define XHASH_SHA_DIGEST_LENGTH 20
 
-# ifndef xhash_NO_DEPRECATED_3_0
 /*-
  * !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
  * ! XHASH_SHA_LONG has to be at least 32 bits wide.                    !
@@ -50,11 +44,9 @@ int xhash_sha1_init(xhash_sha1_ctx_t *c);
 int xhash_sha1_update(xhash_sha1_ctx_t *c, const void *data, size_t len);
 int xhash_sha1_final(unsigned char *md, xhash_sha1_ctx_t *c);
 void xhash_sha1_transform(xhash_sha1_ctx_t *c, const unsigned char *data);
-# endif
 
 unsigned char *xhash_sha1(const unsigned char *d, size_t n, unsigned char *md);
 
-# ifndef xhash_NO_DEPRECATED_3_0
 #  define XHASH_SHA256_CBLOCK   (XHASH_SHA_LBLOCK*4)/* SHA-256 treats input data as a
                                         * contiguous array of 32 bit wide
                                         * big-endian values. */
@@ -67,16 +59,12 @@ typedef struct SHA256state_st {
 } xhash_sha256_ctx_t;
 
 int xhash_sha224_init(xhash_sha256_ctx_t *c);
-int xhash_sha224_update(xhash_sha256_ctx_t *c,
-                                        const void *data, size_t len);
+int xhash_sha224_update(xhash_sha256_ctx_t *c, const void *data, size_t len);
 int xhash_sha224_final(unsigned char *md, xhash_sha256_ctx_t *c);
 int xhash_sha256_init(xhash_sha256_ctx_t *c);
-int xhash_sha256_update(xhash_sha256_ctx_t *c,
-                                        const void *data, size_t len);
+int xhash_sha256_update(xhash_sha256_ctx_t *c, const void *data, size_t len);
 int xhash_sha256_final(unsigned char *md, xhash_sha256_ctx_t *c);
-void xhash_sha256_transform(xhash_sha256_ctx_t *c,
-                                            const unsigned char *data);
-# endif
+void xhash_sha256_transform(xhash_sha256_ctx_t *c, const unsigned char *data);
 
 unsigned char *xhash_sha224(const unsigned char *d, size_t n, unsigned char *md);
 unsigned char *xhash_sha256(const unsigned char *d, size_t n, unsigned char *md);
@@ -87,7 +75,6 @@ unsigned char *xhash_sha256(const unsigned char *d
 # define XHASH_SHA384_DIGEST_LENGTH    48
 # define XHASH_SHA512_DIGEST_LENGTH    64
 
-# ifndef xhash_NO_DEPRECATED_3_0
 /*
  * Unlike 32-bit digest algorithms, SHA-512 *relies* on XHASH_SHA_LONG64
  * being exactly 64-bit wide. See Implementation Notes in sha512.c
@@ -118,16 +105,12 @@ typedef struct SHA512state_st {
 } xhash_sha512_ctx_t;
 
 int xhash_sha384_init(xhash_sha512_ctx_t *c);
-int xhash_sha384_update(xhash_sha512_ctx_t *c,
-                                        const void *data, size_t len);
+int xhash_sha384_update(xhash_sha512_ctx_t *c, const void *data, size_t len);
 int xhash_sha384_final(unsigned char *md, xhash_sha512_ctx_t *c);
 int xhash_sha512_init(xhash_sha512_ctx_t *c);
-int xhash_sha512_update(xhash_sha512_ctx_t *c,
-                                        const void *data, size_t len);
+int xhash_sha512_update(xhash_sha512_ctx_t *c, const void *data, size_t len);
 int xhash_sha512_final(unsigned char *md, xhash_sha512_ctx_t *c);
-void xhash_sha512_transform(xhash_sha512_ctx_t *c,
-                                            const unsigned char *data);
-# endif
+void xhash_sha512_transform(xhash_sha512_ctx_t *c, const unsigned char *data);
 
 unsigned char *xhash_sha384(const unsigned char *d, size_t n, unsigned char *md);
 unsigned char *xhash_sha512(const unsigned char *d, size_t n, unsigned char *md);
