Index: xdigest/include/crypto/md32_common.h
===================================================================
--- xdigest/include/crypto/md32_common.h
+++ xdigest/include/crypto/md32_common.h
@@ -155,7 +155,7 @@
  * Time for some action :-)
  */
 
-int HASH_UPDATE(HASH_CTX *c, const void *data_, size_t len)
+void HASH_UPDATE(HASH_CTX *c, const void *data_, size_t len)
 {
     const unsigned char *data = data_;
     unsigned char *p;
@@ -163,7 +163,7 @@
     size_t n;
 
     if (ossl_unlikely(len == 0))
-        return 1;
+        return;
 
     l = (c->Nl + (((HASH_LONG) len) << 3)) & 0xffffffffUL;
     if (ossl_unlikely(l < c->Nl))              /* overflow */
@@ -193,7 +193,7 @@
         } else {
             memcpy(p + n, data, len);
             c->num += (unsigned int)len;
-            return 1;
+            return;
         }
     }
 
@@ -210,7 +210,6 @@
         c->num = (unsigned int)len;
         memcpy(p, data, len);
     }
-    return 1;
 }
 
 void HASH_TRANSFORM(HASH_CTX *c, const unsigned char *data)
@@ -218,7 +217,7 @@ void HASH_TRANSFORM(HASH_CTX *c, const unsigned ch
     HASH_BLOCK_DATA_ORDER(c, data, 1);
 }
 
-int HASH_FINAL(unsigned char *md, HASH_CTX *c)
+void HASH_FINAL(unsigned char *md, HASH_CTX *c)
 {
     unsigned char *p = (unsigned char *)c->data;
     size_t n = c->num;
@@ -251,8 +250,6 @@ void HASH_TRANSFORM(HASH_CTX *c, const unsigned ch
 # else
     HASH_MAKE_STRING(c, md);
 # endif
-
-    return 1;
 }
 
 # ifndef MD32_REG_T
Index: xdigest/include/crypto/sha.h
===================================================================
--- xdigest/include/crypto/sha.h
+++ xdigest/include/crypto/sha.h
@@ -14,10 +14,10 @@
 
 #include "internal/sha.h"
 
-int ossl_sha256_192_init(xdig_sha256_ctx_t *c);
-int sha512_224_init(xdig_sha512_ctx_t *);
-int sha512_256_init(xdig_sha512_ctx_t *);
-int ossl_sha1_ctrl(xdig_sha1_ctx_t *ctx, int cmd, int mslen, void *ms);
+void ossl_sha256_192_init(xdig_sha256_ctx_t *c);
+void sha512_224_init(xdig_sha512_ctx_t *);
+void sha512_256_init(xdig_sha512_ctx_t *);
+void ossl_sha1_ctrl(xdig_sha1_ctx_t *ctx, int cmd, int mslen, void *ms);
 unsigned char *ossl_sha1(const unsigned char *d, size_t n, unsigned char *md);
 
 #endif
Index: xdigest/include/internal/md2.h
===================================================================
--- xdigest/include/internal/md2.h
+++ xdigest/include/internal/md2.h
@@ -41,10 +41,10 @@ typedef struct MD2state_st {
 #  endif
 #  ifndef xdig_NO_DEPRECATED_3_0
 const char *xdig_md2_ctx_options(void);
-int xdig_md2_ctx_init(xdig_md2_ctx_t *c);
-int xdig_md2_ctx_update(xdig_md2_ctx_t *c, const unsigned char *data,
+void xdig_md2_ctx_init(xdig_md2_ctx_t *c);
+void xdig_md2_ctx_update(xdig_md2_ctx_t *c, const unsigned char *data,
                                      size_t len);
-int xdig_md2_ctx_final(unsigned char *md, xdig_md2_ctx_t *c);
+void xdig_md2_ctx_final(unsigned char *md, xdig_md2_ctx_t *c);
 unsigned char *xdig_md2(const unsigned char *d, size_t n,
                                          unsigned char *md);
 #  endif
Index: xdigest/include/internal/md4.h
===================================================================
--- xdigest/include/internal/md4.h
+++ xdigest/include/internal/md4.h
@@ -47,9 +47,9 @@ typedef struct MD4state_st {
 } xdig_md4_ctx_t;
 #  endif
 #  ifndef xdig_NO_DEPRECATED_3_0
-int xdig_md4_ctx_init(xdig_md4_ctx_t *c);
-int xdig_md4_ctx_update(xdig_md4_ctx_t *c, const void *data, size_t len);
-int xdig_md4_ctx_final(unsigned char *md, xdig_md4_ctx_t *c);
+void xdig_md4_ctx_init(xdig_md4_ctx_t *c);
+void xdig_md4_ctx_update(xdig_md4_ctx_t *c, const void *data, size_t len);
+void xdig_md4_ctx_final(unsigned char *md, xdig_md4_ctx_t *c);
 unsigned char *xdig_md4(const unsigned char *d, size_t n,
                                          unsigned char *md);
 void xdig_md4_ctx_transform(xdig_md4_ctx_t *c, const unsigned char *b);
Index: xdigest/include/internal/md5.h
===================================================================
--- xdigest/include/internal/md5.h
+++ xdigest/include/internal/md5.h
@@ -46,9 +46,9 @@ typedef struct MD5state_st {
 } xdig_md5_ctx_t;
 #  endif
 #  ifndef xdig_NO_DEPRECATED_3_0
-int xdig_md5_ctx_init(xdig_md5_ctx_t *c);
-int xdig_md5_ctx_update(xdig_md5_ctx_t *c, const void *data, size_t len);
-int xdig_md5_ctx_final(unsigned char *md, xdig_md5_ctx_t *c);
+void xdig_md5_ctx_init(xdig_md5_ctx_t *c);
+void xdig_md5_ctx_update(xdig_md5_ctx_t *c, const void *data, size_t len);
+void xdig_md5_ctx_final(unsigned char *md, xdig_md5_ctx_t *c);
 unsigned char *xdig_md5(const unsigned char *d, size_t n,
                                          unsigned char *md);
 void xdig_md5_ctx_transform(xdig_md5_ctx_t *c, const unsigned char *b);
Index: xdigest/include/internal/sha.h
===================================================================
--- xdigest/include/internal/sha.h
+++ xdigest/include/internal/sha.h
@@ -46,9 +46,9 @@ typedef struct SHAstate_st {
     unsigned int num;
 } xdig_sha1_ctx_t;
 
-int xdig_sha1_ctx_init(xdig_sha1_ctx_t *c);
-int xdig_sha1_ctx_update(xdig_sha1_ctx_t *c, const void *data, size_t len);
-int xdig_sha1_ctx_final(unsigned char *md, xdig_sha1_ctx_t *c);
+void xdig_sha1_ctx_init(xdig_sha1_ctx_t *c);
+void xdig_sha1_ctx_update(xdig_sha1_ctx_t *c, const void *data, size_t len);
+void xdig_sha1_ctx_final(unsigned char *md, xdig_sha1_ctx_t *c);
 void xdig_sha1_ctx_transform(xdig_sha1_ctx_t *c, const unsigned char *data);
 # endif
 
@@ -66,14 +66,14 @@ typedef struct SHA256state_st {
     unsigned int num, md_len;
 } xdig_sha256_ctx_t;
 
-int xdig_sha224_ctx_init(xdig_sha256_ctx_t *c);
-int xdig_sha224_ctx_update(xdig_sha256_ctx_t *c,
+void xdig_sha224_ctx_init(xdig_sha256_ctx_t *c);
+void xdig_sha224_ctx_update(xdig_sha256_ctx_t *c,
                                         const void *data, size_t len);
-int xdig_sha224_ctx_final(unsigned char *md, xdig_sha256_ctx_t *c);
-int xdig_sha256_ctx_init(xdig_sha256_ctx_t *c);
-int xdig_sha256_ctx_update(xdig_sha256_ctx_t *c,
+void xdig_sha224_ctx_final(unsigned char *md, xdig_sha256_ctx_t *c);
+void xdig_sha256_ctx_init(xdig_sha256_ctx_t *c);
+void xdig_sha256_ctx_update(xdig_sha256_ctx_t *c,
                                         const void *data, size_t len);
-int xdig_sha256_ctx_final(unsigned char *md, xdig_sha256_ctx_t *c);
+void xdig_sha256_ctx_final(unsigned char *md, xdig_sha256_ctx_t *c);
 void xdig_sha256_ctx_transform(xdig_sha256_ctx_t *c,
                                             const unsigned char *data);
 # endif
@@ -117,14 +117,14 @@ typedef struct SHA512state_st {
     unsigned int num, md_len;
 } xdig_sha512_ctx_t;
 
-int xdig_sha384_ctx_init(xdig_sha512_ctx_t *c);
-int xdig_sha384_ctx_update(xdig_sha512_ctx_t *c,
+void xdig_sha384_ctx_init(xdig_sha512_ctx_t *c);
+void xdig_sha384_ctx_update(xdig_sha512_ctx_t *c,
                                         const void *data, size_t len);
-int xdig_sha384_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c);
-int xdig_sha512_ctx_init(xdig_sha512_ctx_t *c);
-int xdig_sha512_ctx_update(xdig_sha512_ctx_t *c,
+void xdig_sha384_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c);
+void xdig_sha512_ctx_init(xdig_sha512_ctx_t *c);
+void xdig_sha512_ctx_update(xdig_sha512_ctx_t *c,
                                         const void *data, size_t len);
-int xdig_sha512_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c);
+void xdig_sha512_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c);
 void xdig_sha512_ctx_transform(xdig_sha512_ctx_t *c,
                                             const unsigned char *data);
 # endif
Index: xdigest/md2/md2_dgst.c
===================================================================
--- xdigest/md2/md2_dgst.c
+++ xdigest/md2/md2_dgst.c
@@ -75,22 +75,18 @@ const char *xdig_md2_ctx_options(void)
         return "md2(int)";
 }
 
-int xdig_md2_ctx_init(xdig_md2_ctx_t *c)
+void xdig_md2_ctx_init(xdig_md2_ctx_t *c)
 {
     c->num = 0;
     memset(c->state, 0, sizeof(c->state));
     memset(c->cksm, 0, sizeof(c->cksm));
     memset(c->data, 0, sizeof(c->data));
-    return 1;
 }
 
-int xdig_md2_ctx_update(xdig_md2_ctx_t *c, const unsigned char *data, size_t len)
+void xdig_md2_ctx_update(xdig_md2_ctx_t *c, const unsigned char *data, size_t len)
 {
     register UCHAR *p;
 
-    if (len == 0)
-        return 1;
-
     p = c->data;
     if (c->num != 0) {
         if ((c->num + len) >= MD2_BLOCK) {
@@ -104,7 +100,7 @@ const char *xdig_md2_ctx_options(void)
             memcpy(&(p[c->num]), data, len);
             /* data+=len; */
             c->num += (int)len;
-            return 1;
+            return;
         }
     }
     /*
@@ -118,7 +114,6 @@ const char *xdig_md2_ctx_options(void)
     }
     memcpy(p, data, len);
     c->num = (int)len;
-    return 1;
 }
 
 static void md2_block(xdig_md2_ctx_t *c, const unsigned char *d)
@@ -154,7 +149,7 @@ static void md2_block(xdig_md2_ctx_t *c, const uns
     xdig_cleanse(state, 48 * sizeof(MD2_INT));
 }
 
-int xdig_md2_ctx_final(unsigned char *md, xdig_md2_ctx_t *c)
+void xdig_md2_ctx_final(unsigned char *md, xdig_md2_ctx_t *c)
 {
     int i, v;
     register UCHAR *cp;
@@ -176,5 +171,4 @@ static void md2_block(xdig_md2_ctx_t *c, const uns
     for (i = 0; i < 16; i++)
         md[i] = (UCHAR) (p1[i] & 0xff);
     xdig_cleanse(c, sizeof(*c));
-    return 1;
 }
Index: xdigest/md4/md4_dgst.c
===================================================================
--- xdigest/md4/md4_dgst.c
+++ xdigest/md4/md4_dgst.c
@@ -26,7 +26,7 @@
 #define INIT_DATA_C (unsigned long)0x98badcfeL
 #define INIT_DATA_D (unsigned long)0x10325476L
 
-int xdig_md4_ctx_init(xdig_md4_ctx_t *c)
+void xdig_md4_ctx_init(xdig_md4_ctx_t *c)
 {
     memset(c, 0, sizeof(*c));
     c->A = INIT_DATA_A;
@@ -33,7 +33,6 @@
     c->B = INIT_DATA_B;
     c->C = INIT_DATA_C;
     c->D = INIT_DATA_D;
-    return 1;
 }
 
 #ifndef md4_block_data_order
Index: xdigest/md5/md5_dgst.c
===================================================================
--- xdigest/md5/md5_dgst.c
+++ xdigest/md5/md5_dgst.c
@@ -26,7 +26,7 @@
 #define INIT_DATA_C (unsigned long)0x98badcfeL
 #define INIT_DATA_D (unsigned long)0x10325476L
 
-int xdig_md5_ctx_init(xdig_md5_ctx_t *c)
+void xdig_md5_ctx_init(xdig_md5_ctx_t *c)
 {
     memset(c, 0, sizeof(*c));
     c->A = INIT_DATA_A;
@@ -33,7 +33,6 @@
     c->B = INIT_DATA_B;
     c->C = INIT_DATA_C;
     c->D = INIT_DATA_D;
-    return 1;
 }
 
 #ifndef md5_block_data_order
Index: xdigest/sha/sha256.c
===================================================================
--- xdigest/sha/sha256.c
+++ xdigest/sha/sha256.c
@@ -13,6 +13,7 @@
  */
 /* ignored include 'internal/deprecated.h' */
 
+
 /* ignored include 'openssl/opensslconf.h' */
 
 #include <stdlib.h>
@@ -24,7 +25,7 @@
 #include "internal/endian.h"
 #include "crypto/sha.h"
 
-int xdig_sha224_ctx_init(xdig_sha256_ctx_t *c)
+void xdig_sha224_ctx_init(xdig_sha256_ctx_t *c)
 {
     memset(c, 0, sizeof(*c));
     c->h[0] = 0xc1059ed8UL;
@@ -36,10 +37,9 @@
     c->h[6] = 0x64f98fa7UL;
     c->h[7] = 0xbefa4fa4UL;
     c->md_len = XDIG_SHA224_DIGEST_LENGTH;
-    return 1;
 }
 
-int xdig_sha256_ctx_init(xdig_sha256_ctx_t *c)
+void xdig_sha256_ctx_init(xdig_sha256_ctx_t *c)
 {
     memset(c, 0, sizeof(*c));
     c->h[0] = 0x6a09e667UL;
@@ -51,7 +51,6 @@
     c->h[6] = 0x1f83d9abUL;
     c->h[7] = 0x5be0cd19UL;
     c->md_len = XDIG_SHA256_DIGEST_LENGTH;
-    return 1;
 }
 
 unsigned char *xdig_sha256(const unsigned char *d, size_t n, unsigned char *md)
@@ -65,21 +64,20 @@ unsigned char *xdig_sha256(const unsigned char *d,
     return md;
 }
 
-int ossl_sha256_192_init(xdig_sha256_ctx_t *c)
+void ossl_sha256_192_init(xdig_sha256_ctx_t *c)
 {
     xdig_sha256_ctx_init(c);
     c->md_len = XDIG_SHA256_192_DIGEST_LENGTH;
-    return 1;
 }
 
-int xdig_sha224_ctx_update(xdig_sha256_ctx_t *c, const void *data, size_t len)
+void xdig_sha224_ctx_update(xdig_sha256_ctx_t *c, const void *data, size_t len)
 {
-    return xdig_sha256_ctx_update(c, data, len);
+    xdig_sha256_ctx_update(c, data, len);
 }
 
-int xdig_sha224_ctx_final(unsigned char *md, xdig_sha256_ctx_t *c)
+void xdig_sha224_ctx_final(unsigned char *md, xdig_sha256_ctx_t *c)
 {
-    return xdig_sha256_ctx_final(md, c);
+    xdig_sha256_ctx_final(md, c);
 }
 
 unsigned char *xdig_sha224(const unsigned char *d, size_t n, unsigned char *md)
@@ -137,8 +135,6 @@ size_t xdig_sha224_ctx_size(void)
                 }                       \
                 break;                  \
             default:                    \
-                if ((c)->md_len > XDIG_SHA256_DIGEST_LENGTH) \
-                    return 0;                           \
                 for (nn=0;nn<(c)->md_len/4;nn++) {              \
                     ll=(c)->h[nn]; (void)HOST_l2c(ll,(s));      \
                 }                       \
Index: xdigest/sha/sha512.c
===================================================================
--- xdigest/sha/sha512.c
+++ xdigest/sha/sha512.c
@@ -75,7 +75,7 @@
 # define U64(C)     C##ULL
 #endif
 
-int sha512_224_init(xdig_sha512_ctx_t *c)
+void sha512_224_init(xdig_sha512_ctx_t *c)
 {
     c->h[0] = U64(0x8c3d37c819544da2);
     c->h[1] = U64(0x73e1996689dcd4d6);
@@ -90,10 +90,9 @@
     c->Nh = 0;
     c->num = 0;
     c->md_len = XDIG_SHA224_DIGEST_LENGTH;
-    return 1;
 }
 
-int sha512_256_init(xdig_sha512_ctx_t *c)
+void sha512_256_init(xdig_sha512_ctx_t *c)
 {
     c->h[0] = U64(0x22312194fc2bf72c);
     c->h[1] = U64(0x9f555fa3c84c64c2);
@@ -108,10 +107,9 @@
     c->Nh = 0;
     c->num = 0;
     c->md_len = XDIG_SHA256_DIGEST_LENGTH;
-    return 1;
 }
 
-int xdig_sha384_ctx_init(xdig_sha512_ctx_t *c)
+void xdig_sha384_ctx_init(xdig_sha512_ctx_t *c)
 {
     c->h[0] = U64(0xcbbb9d5dc1059ed8);
     c->h[1] = U64(0x629a292a367cd507);
@@ -126,7 +124,6 @@
     c->Nh = 0;
     c->num = 0;
     c->md_len = XDIG_SHA384_DIGEST_LENGTH;
-    return 1;
 }
 
 unsigned char *xdig_sha384(const unsigned char *d, size_t n, unsigned char *md)
@@ -140,7 +137,7 @@ unsigned char *xdig_sha384(const unsigned char *d,
     return md;
 }
 
-int xdig_sha512_ctx_init(xdig_sha512_ctx_t *c)
+void xdig_sha512_ctx_init(xdig_sha512_ctx_t *c)
 {
     c->h[0] = U64(0x6a09e667f3bcc908);
     c->h[1] = U64(0xbb67ae8584caa73b);
@@ -155,7 +152,6 @@ unsigned char *xdig_sha384(const unsigned char *d,
     c->Nh = 0;
     c->num = 0;
     c->md_len = XDIG_SHA512_DIGEST_LENGTH;
-    return 1;
 }
 
 unsigned char *xdig_sha512(const unsigned char *d, size_t n, unsigned char *md)
@@ -188,7 +184,7 @@ void sha512_block_data_order_c(xdig_sha512_ctx_t *
 #endif
 void sha512_block_data_order(xdig_sha512_ctx_t *ctx, const void *in, size_t num);
 
-int xdig_sha512_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c)
+void xdig_sha512_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c)
 {
     unsigned char *p = (unsigned char *)c->u.p;
     size_t n = c->num;
@@ -226,9 +222,6 @@ void sha512_block_data_order(xdig_sha512_ctx_t *ct
 
     sha512_block_data_order(c, p, 1);
 
-    if (md == 0)
-        return 0;
-
     switch (c->md_len) {
     /* Let compiler decide if it's appropriate to unroll... */
     case XDIG_SHA224_DIGEST_LENGTH:
@@ -301,26 +294,21 @@ void sha512_block_data_order(xdig_sha512_ctx_t *ct
         break;
     /* ... as well as make sure md_len is not abused. */
     default:
-        return 0;
+        abort(); /* we should never end up here */
     }
-
-    return 1;
 }
 
-int xdig_sha384_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c)
+void xdig_sha384_ctx_final(unsigned char *md, xdig_sha512_ctx_t *c)
 {
-    return xdig_sha512_ctx_final(md, c);
+    xdig_sha512_ctx_final(md, c);
 }
 
-int xdig_sha512_ctx_update(xdig_sha512_ctx_t *c, const void *_data, size_t len)
+void xdig_sha512_ctx_update(xdig_sha512_ctx_t *c, const void *_data, size_t len)
 {
     XDIG_SHA_LONG64 l;
     unsigned char *p = c->u.p;
     const unsigned char *data = (const unsigned char *)_data;
 
-    if (len == 0)
-        return 1;
-
     l = (c->Nl + (((XDIG_SHA_LONG64) len) << 3)) & U64(0xffffffffffffffff);
     if (l < c->Nl)
         c->Nh++;
@@ -333,7 +321,7 @@ void sha512_block_data_order(xdig_sha512_ctx_t *ct
 
         if (len < n) {
             memcpy(p + c->num, data, len), c->num += (unsigned int)len;
-            return 1;
+            return;
         } else {
             memcpy(p + c->num, data, n), c->num = 0;
             len -= n, data += n;
@@ -356,13 +344,11 @@ void sha512_block_data_order(xdig_sha512_ctx_t *ct
 
     if (len != 0)
         memcpy(p, data, len), c->num = (int)len;
-
-    return 1;
 }
 
-int xdig_sha384_ctx_update(xdig_sha512_ctx_t *c, const void *data, size_t len)
+void xdig_sha384_ctx_update(xdig_sha512_ctx_t *c, const void *data, size_t len)
 {
-    return xdig_sha512_ctx_update(c, data, len);
+    xdig_sha512_ctx_update(c, data, len);
 }
 
 void xdig_sha512_ctx_transform(xdig_sha512_ctx_t *c, const unsigned char *data)
Index: xdigest/sha/sha_local.h
===================================================================
--- xdigest/sha/sha_local.h
+++ xdigest/sha/sha_local.h
@@ -51,7 +51,7 @@ void sha1_block_data_order(xdig_sha1_ctx_t *c, con
 #define INIT_DATA_h3 0x10325476UL
 #define INIT_DATA_h4 0xc3d2e1f0UL
 
-int HASH_INIT(xdig_sha1_ctx_t *c)
+void HASH_INIT(xdig_sha1_ctx_t *c)
 {
     memset(c, 0, sizeof(*c));
     c->h0 = INIT_DATA_h0;
@@ -59,7 +59,6 @@ void sha1_block_data_order(xdig_sha1_ctx_t *c, con
     c->h2 = INIT_DATA_h2;
     c->h3 = INIT_DATA_h3;
     c->h4 = INIT_DATA_h4;
-    return 1;
 }
 
 #define K_00_19 0x5a827999UL
