Index: include/internal/common.h
===================================================================
--- include/internal/common.h
+++ include/internal/common.h
@@ -63,6 +63,8 @@ __owur static ossl_inline int ossl_assert_int(int
 
 # endif
 
+void xdig_cleanse(void *ptr, size_t len);
+
 /* Check if |pre|, which must be a string literal, is a prefix of |str| */
 #define HAS_PREFIX(str, pre) (strncmp(str, pre "", sizeof(pre) - 1) == 0)
 /* As before, and if check succeeds, advance |str| past the prefix |pre| */
Index: xdigest/md5/md5_one.c
===================================================================
--- xdigest/md5/md5_one.c
+++ xdigest/md5/md5_one.c
@@ -17,6 +17,7 @@
 #include <string.h>
 #include <xdigest/md5.h>
 /* ignored include 'openssl/crypto.h' */
+#include "internal/common.h"
 
 #ifdef CHARSET_EBCDIC
 # include <xdigest/ebcdic.h>
Index: xdigest/md4/md4_one.c
===================================================================
--- xdigest/md4/md4_one.c
+++ xdigest/md4/md4_one.c
@@ -17,6 +17,7 @@
 #include <string.h>
 #include <xdigest/md4.h>
 /* ignored include 'openssl/crypto.h' */
+#include "internal/common.h"
 
 #ifdef CHARSET_EBCDIC
 # include <xdigest/ebcdic.h>
Index: xdigest/md2/md2_dgst.c
===================================================================
--- xdigest/md2/md2_dgst.c
+++ xdigest/md2/md2_dgst.c
@@ -19,6 +19,7 @@
 #include <xdigestest/md2.h>
 /* ignored include 'openssl/opensslv.h' */
 /* ignored include 'openssl/crypto.h' */
+#include "internal/common.h"
 
 /*
  * Implemented from RFC1319 The MD2 Message-Digest Algorithm
Index: xdigest/md2/md2_one.c
===================================================================
--- xdigest/md2/md2_one.c
+++ xdigest/md2/md2_one.c
@@ -16,6 +16,7 @@
 #include <stdio.h>
 /* ignored include 'internal/cryptlib.h' */
 #include "internal/md2.h"
+#include "internal/common.h"
 
 /*
  * This is a separate file so that #defines in cryptlib.h can map my MD
